### ISSUE TITLE
Add connectivity-based mode to compute trans. probs for last time point in WOT kernel

### ISSUE THREAD

**Issue by Marius1311** [theislab/cellrank#608]
- [x] Additional function parameters, change functionality or change defaults?
- [ ] New estimator in `cr.tl.estimators`?
- [ ] New kernel in `cr.tl.kernels`?
- [ ] New gene trend model in `cr.ul.models`?
- [ ] New plotting function in `cr.pl`?
- [ ] Other?

For the last time point, we should have three modes:
- `diagonal`
- `uniform`
- `connectivities`

The last one should internally subset the AnnData to the last time-point, re-compute the KNN graph, compute a connectivity-based transition matrix based on that, and combine that with the output of the WOT kernel.

**Comment by Marius1311** [comment]
Here's some motivation for doing this: I'm using the external `WOTKernel` to compute transition probabilities, where I assign the identity to the last time point (sparse diagonal). That implies that all cells from the last time point are absorbing states in the Markov chain. I would like to compute absorption probabilities, so I manually define some terminal states, see Fig. 1. However, these are not the only absorbing states here, as all day-18 (last time point) cells are absorbing. When I compute absorption probs towards these terminal states, I get

```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-48-6ddbc3fab9e4> in <module>
----> 1 g.compute_absorption_probabilities(n_jobs=8, solver='gmres', use_petsc='True')

~/Projects/cellrank/cellrank/tl/estimators/_base_estimator.py in compute_absorption_probabilities(self, keys, check_irreducibility, solver, use_petsc, time_to_absorption, n_jobs, backend, show_progress_bar, tol, preconditioner)
    575         mask = np.isclose(abs_classes.sum(1), 1.0, rtol=1e-3)
    576         if not np.all(mask):
--> 577             raise ValueError(
    578                 f"`{np.sum(~mask)}` value(s) do not sum to 1 (rtol=1e-3). Try decreasing the tolerance as `tol=...`, "
    579                 f"specifying a preconditioner as `preconditioner=...` or "

ValueError: `40162` value(s) do not sum to 1 (rtol=1e-3). Try decreasing the tolerance as `tol=...`, specifying a preconditioner as `preconditioner=...` or use a direct solver as `solver='direct'` if the matrix is small.
```

Which makes a lot of sense, and is actually the correct output here. This case just doesn't fit into our usual framework, because usually, only cells in terminal states are absorbing. 

## Fig. 1
Terminal states. Only contain day 18 cells. 
![image](https://user-images.githubusercontent.com/24494650/121178285-9c518000-c85e-11eb-8034-6ccc16fb322e.png)

**Comment by michalk8** [comment]
closed via #609
